### DISCUSSION TITLE
Mosquitto post-install info

### DISCUSSION THREAD

**Discussion by havardthom** [community-scripts/ProxmoxVE#782] [Guides]
Original thread: https://github.com/tteck/Proxmox/discussions/2921

Mosquitto comes with a password file generating utility called mosquitto_passwd.

```
mosquitto_passwd -c /etc/mosquitto/passwd <user>
```

Ensure that the file is managed by the appropriate user and group.

```
chown mosquitto:mosquitto /etc/mosquitto/passwd
```

Now restart Mosquitto server.

```
systemctl restart mosquitto
```

The command below will show what ports are listening (look for *:1883 (LISTEN))

```
lsof -i -P -n | grep LISTEN
```

**Comment by aktentasche** [comment]
Alternatively disable authentication by running `nano /etc/mosquitto/conf.d/default.conf` and changing:

```
allow_anonymous false
persistence true
password_file /etc/mosquitto/passwd
listener 1883
```

to 

```
allow_anonymous true
persistence true
# password_file /etc/mosquitto/passwd
listener 1883
```

**Comment by agautam** [comment]
> Original thread: [tteck/Proxmox#2921](https://github.com/tteck/Proxmox/discussions/2921)
> 
> Mosquitto comes with a password file generating utility called mosquitto_passwd.
> 
> ```
> mosquitto_passwd -c /etc/mosquitto/passwd <user>
> ```
> 
> Ensure that the file is managed by the appropriate user and group.
> 
> ```
> chown mosquitto:mosquitto /etc/mosquitto/passwd
> ```
Small recommendation here - make the file owned by user=root and group=mosquitto.
```
chown root:mosquitto /etc/mosquitto/passwd
```
Then update the permissions so it may be read by the group
```
chmod 0640 /etc/mosquitto/passwd
```
An additional note here:
`mosquitto_passwd` utility WILL give a warning if the file isn't owned by `root:root` and/or its permissions are anything wider than `0700`. Unfortunately with that permission set and ownership - the file isn't readable by `mosquitto service` itself. So... pick your poison.

**Reply by danielo515** [reply]
Mosquitto is giving a warning that  the password file must be owned by mosquitto or it will not be loaded in future versions
